src/Articulate/UmbracoHelperExtensions.cs

```diff
@@ -406,8 +407,81 @@ private static IPublishedContent[] GetListNodes(IMasterModel masterModel)
             }
             return listNodes;
         }
-        
-        
+
+        /// <summary>
+        /// Get all recent blog articles sorted
+        /// </summary>
+        /// <param name="helper"></param>
+        /// <param name="id"></param>
+        /// <param name="sortType"></param>
+        /// <returns>IEnumerable<PostModel></returns>
+        public static IEnumerable<PostModel> GetSortedAllPosts(UmbracoHelper helper, int id, string sortType)
+        {           
+            IContentService contentService = Current.Services.ContentService;
+
+            var publishedContent = new MasterModel(helper.Content(id));
+
+            var listNodes = publishedContent.RootBlogNode.Children("ArticulateArchive").ToArray();
+
+            var listNodeIds = listNodes.Select(x => x.Id).ToArray();
+
+            var totalPosts = GetPostCount(helper, listNodes.Select(x => x.Id).ToArray());
+
+            var pager = new PagerModel(totalPosts, 0, 1);
+
+            var listItems = helper.GetPostsSortedByPublishedDate(pager, null, listNodeIds);
+            var posts = new ListModel(listNodes[0], listItems, pager).Posts;
+
+            List<PostModel> PublishedPostList = new List<PostModel>();
+            List<PostModel> UnPublishedPostList = new List<PostModel>();
+
+            if (sortType == "recent-edited-published")
+            {
+                foreach (var (nodes, content) in from nodes in posts
```

Comment:
This is going to be tremendously slow and have big performance implications. You are using the IContentService which is going to go directly to the database. This is also an N+1 query problem so for each IPublishedContent item you are performing a service lookup which can be several db queries each. You should be able to determine published or not just from the IPublishedContent that is already resolved.